Added Artron Inhibitor

NovaAstral opened the pull request:
- Added Artron Inhibitor, an entity like the Time Distortion Generator, except it will slowly drain the Artron Energy from any TARDIS within its radius

Note: Related Sonic Screwdriver PR: https://github.com/MattJeanes/Sonic-Screwdriver/pull/41

NovaAstral committed:
Added Artron Inhibitor
- Added Artron Inhibitor, an entity like the Time Distortion Generator, except it will slowly drain the Artron Energy from any TARDIS within its radius

NovaAstral committed:
fix comments

AmyJeanes committed:
Merge remote-tracking branch 'origin/dev' into artron-inhibitor

AmyJeanes commented on lua/entities/gmod_artron_inhibitor/shared.lua:
```diff
@@ -0,0 +1,17 @@
+ENT.Type = "anim"
+ENT.Base = "base_gmodentity"
+
+ENT.PrintName = "#TARDIS.ArtronInhibitor"
```
There are missing translations for the following strings
- `ArtronInhibitor`
- `ArtronInhibitor.Instructions`
- `ArtronInhibitor.Starting`
- `ArtronInhibitor.Disabled`
- `ArtronInhibitor.Enabled`

These should be added into the `i18n/en.json` file and run the Generate Languages task in VSCode (Ctrl+Shift+P -> Run task -> Generate Languages)

Note you need PowerShell Core installed for this to work

NovaAstral replied:
Added language but I don't have PowerShell Core and I don't know which one is the correct one.

AmyJeanes replied:
Looks good! Note I've made some small tweaks to match the TDG language

AmyJeanes commented on lua/entities/gmod_artron_inhibitor/init.lua:
```diff
@@ -0,0 +1,214 @@
+        self.LastActivator = nil
+
+        self:GetPhysicsObject():SetVelocity(Vector(0, 0, 0))
+        self:SetColor(Color(164, 90, 250, self:GetColor().a))
```
It would make sense to have this be a different colour to the TDG

NovaAstral replied:
started color is now orange (the startup color of the tdg because annoyingly its a good color for artron energy)
startup color is blue

AmyJeanes commented on lua/entities/gmod_artron_inhibitor/init.lua:
```diff
@@ -0,0 +1,214 @@
+    return self.Radius
+end
+
+function ENT:Repair(repair)
```
We would also want to add this into the sonic screwdriver so it can be repaired etc, can you make another PR for that?

I've added you as a collaborator to that repo now

NovaAstral replied:
made a pr in the sonicsd for this

AmyJeanes reviewed (changes requested):
This looks like a useful roleplay tool, nice work! We're missing translations and a couple other tiny things but looks good to me overall.

NovaAstral committed:
artron inhibitor color change and language
- Changed startup color to blue
- Changed started color to orange/yellow
- Added en language (Don't have powershell core and don't know which one it is)

NovaAstral committed:
Update Generated Languages

AmyJeanes committed:
wording tweaks

AmyJeanes reviewed (approved):
Looks good from my perspective, but I also want @parar020100 to have a look at this as well

AmyJeanes committed:
Merge remote-tracking branch 'origin/dev' into artron-inhibitor

parar020100 reviewed (approved):
Looks good to me, nice job!
